### PR TITLE
[CI] Improve issue templates

### PR DESCRIPTION
**Pull request by KornevNikita**
Use YAML & add new bug templates to improve issue triaging.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/1-bug-report.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      required: true
+  - type: textarea
+    id: how-to-reproduce
+    attributes:
+      label: To reproduce
+      description: Please describe the steps to reproduce the behavior
+      placeholder: |
+        1. Include code snippet as short as possible
+        2. Specify the command which should be used to compile the program
+        3. Specify the comment which should be used to launch the program
```
Comment (dm-vodopyanov) [inline .github/ISSUE_TEMPLATE/1-bug-report.yml:20]:
```suggestion
        3. Specify the command which should be used to launch the program
```

File: .github/ISSUE_TEMPLATE/1-bug-report.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  - type: textarea
+    id: environment
+    attributes:
+      label: Environment
+      description: Please complete the following information
+      placeholder: |
+        - OS: [e.g Windows/Linux]
+        - Target device and vendor: [e.g. Intel GPU]
+        - DPC++ version: [e.g. commit hash or output of `clang++ --version`]
+        - Dependencies version: [e.g. low-level runtime versions (like NEO 20.04)]
```
Comment (dm-vodopyanov) [inline .github/ISSUE_TEMPLATE/1-bug-report.yml:33]: Maybe instead ask customers to provide the output of `sycl-ls --version` command?

File: .github/ISSUE_TEMPLATE/2-bug-report-cuda.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      required: true
+  - type: textarea
+    id: how-to-reproduce
+    attributes:
+      label: To reproduce
+      description: Please describe the steps to reproduce the behavior
+      placeholder: |
+        1. Include code snippet as short as possible
+        2. Specify the command which should be used to compile the program
+        3. Specify the comment which should be used to launch the program
```
Comment (dm-vodopyanov) [inline .github/ISSUE_TEMPLATE/2-bug-report-cuda.yml:20]:
```suggestion
        3. Specify the command which should be used to launch the program
```

File: .github/ISSUE_TEMPLATE/2-bug-report-cuda.yml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    validations:
+      required: false
+  - type: textarea
+    id: environment
+    attributes:
+      label: Environment
+      description: Please complete the following information
+      placeholder: |
+        - OS: [e.g Windows/Linux]
+        - Target device and vendor: [e.g. Intel GPU]
```
Comment (dm-vodopyanov) [inline .github/ISSUE_TEMPLATE/2-bug-report-cuda.yml:31]: Maybe specify the example for this template?
```suggestion
        - Target device and vendor: [e.g. Nvidia GPU]
```

File: .github/ISSUE_TEMPLATE/2-bug-report-cuda.yml [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+        - DPC++ version: [e.g. commit hash or output of `clang++ --version`]
+        - Dependencies version: [e.g. low-level runtime versions (like NEO 20.04)]
```
Comment (dm-vodopyanov) [inline .github/ISSUE_TEMPLATE/2-bug-report-cuda.yml:33]: In addition to comment above about changing to `sycl-ls --version` command, maybe ask also to provide the output of `nvidia-smi`?

File: .github/ISSUE_TEMPLATE/3-bug-report-hip.yml [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+      required: true
+  - type: textarea
+    id: how-to-reproduce
+    attributes:
+      label: To reproduce
+      description: Please describe the steps to reproduce the behavior
+      placeholder: |
+        1. Include code snippet as short as possible
+        2. Specify the command which should be used to compile the program
+        3. Specify the comment which should be used to launch the program
```
Comment (dm-vodopyanov) [inline .github/ISSUE_TEMPLATE/3-bug-report-hip.yml:20]:
```suggestion
        3. Specify the command which should be used to launch the program
```

File: .github/ISSUE_TEMPLATE/3-bug-report-hip.yml [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+    validations:
+      required: false
+  - type: textarea
+    id: environment
+    attributes:
+      label: Environment
+      description: Please complete the following information
+      placeholder: |
+        - OS: [e.g Windows/Linux]
+        - Target device and vendor: [e.g. Intel GPU]
```
Comment (dm-vodopyanov) [inline .github/ISSUE_TEMPLATE/3-bug-report-hip.yml:31]: Maybe specify the example for this template?
```suggestion
        - Target device and vendor: [e.g. AMD GPU]
```

File: .github/ISSUE_TEMPLATE/3-bug-report-hip.yml [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+        - DPC++ version: [e.g. commit hash or output of `clang++ --version`]
+        - Dependencies version: [e.g. low-level runtime versions (like NEO 20.04)]
```
Comment (dm-vodopyanov) [inline .github/ISSUE_TEMPLATE/3-bug-report-hip.yml:33]: In addition to comment above about changing to `sycl-ls --version` command, maybe ask also to provide the output of `rocm-smi`?

File: .github/ISSUE_TEMPLATE/1-bug-report.yml [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
```
Comment (KornevNikita) [inline .github/ISSUE_TEMPLATE/1-bug-report.yml:33]:
```suggestion
        - Dependencies version: [e.g. the output of `sycl-ls`]
```

File: .github/ISSUE_TEMPLATE/3-bug-report-hip.yml [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+  - type: textarea
+    id: environment
+    attributes:
+      label: Environment
+      description: Please complete the following information
+      placeholder: |
+        - OS: [e.g Windows/Linux]
+        - Target device and vendor: [e.g. AMD GPU]
+        - DPC++ version: [e.g. commit hash or output of `clang++ --version`]
+        - Dependencies version: [e.g. low-level runtime versions (like NEO 20.04)]
```
Comment (KornevNikita) [inline .github/ISSUE_TEMPLATE/3-bug-report-hip.yml:33]:
```suggestion
        - Dependencies version: [e.g. rocm-smi]
```

**Review by aelovikov-intel — APPROVED** [review]
As long as there is a possibility to submit a "blank issue" I'm fine. As for the templates, I think the people who work with public issues more than I should be reviewing/approving this.

I can give the formal approval so that you can merge it once reviewed by others though.

File: .github/ISSUE_TEMPLATE/1-bug-report.yml [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+      description: A clear and concise description of what the bug is.
+    validations:
+      required: true
+  - type: textarea
+    id: how-to-reproduce
+    attributes:
+      label: To reproduce
+      description: Please describe the steps to reproduce the behavior
+      placeholder: |
+        1. Include code snippet as short as possible
```
Comment (jzc) [inline .github/ISSUE_TEMPLATE/1-bug-report.yml:18]:
```suggestion
        1. Include a code snippet that is as short as possible
```

File: .github/ISSUE_TEMPLATE/1-bug-report.yml [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+  - type: textarea
+    id: environment
+    attributes:
+      label: Environment
+      description: Please complete the following information
+      placeholder: |
+        - OS: [e.g Windows/Linux]
+        - Target device and vendor: [e.g. Intel GPU]
+        - DPC++ version: [e.g. commit hash or output of `clang++ --version`]
+        - Dependencies version: [e.g. the output of `sycl-ls`]
```
Comment (KornevNikita) [inline .github/ISSUE_TEMPLATE/1-bug-report.yml:33]:
```suggestion
        - Dependencies version: [e.g. the output of `sycl-ls --verbose`]
```

File: .github/ISSUE_TEMPLATE/3-bug-report-hip.yml [hunk 13]
```diff
@@ -0,0 +1,42 @@
...
+  - type: textarea
+    id: environment
+    attributes:
+      label: Environment
+      description: Please complete the following information
+      placeholder: |
+        - OS: [e.g Windows/Linux]
+        - Target device and vendor: [e.g. AMD GPU]
+        - DPC++ version: [e.g. commit hash or output of `clang++ --version`]
+        - Dependencies version: [e.g. the output of `rocm-smi`]
```
Comment (KornevNikita) [inline .github/ISSUE_TEMPLATE/3-bug-report-hip.yml:33]:
```suggestion
        - Dependencies version: [e.g. the output of `rocm-smi` and `sycl-ls --verbose`]
```

**Review by dm-vodopyanov — APPROVED** [review]

**Pull request merged**